api/controllers/profileController.js

```diff
@@ -0,0 +1,102 @@
+const Profile = require("../models/profileModel");
+
+const getAllProfiles = async (req, res) => {
+  try {
+    const profiles = await Profile.find({});
+    res.status(200).json(profiles);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+const getProfileById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const profile = await Profile.findById(id);
+    if (!profile) {
+      return res.status(404).json({ message: "Profile not found" });
+    }
+    res.status(200).json(profile);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+const createProfile = async (req, res) => {
+  try {
+    const { accountRef, profileId } = req.body;
+
+    if (accountRef === "" || profileId === "" || !accountRef || !profileId) {
+      return res
+        .status(400)
+        .json({ message: "Please fill in all required fields" });
+    }
+
+    if (await Profile.findOne({ profileId })) {
+      return res.status(409).json({ message: "Profile with this ID already exists" });
+    }
+
+    profileName = profileId;
+
+    const newProfile = new Profile({
+      accountRef,
+      profileId,
+      profileName,
+    });
+
+    const savedProfile = await newProfile.save();
+    res.status(201).json(savedProfile); // 201 Created status code
+  } catch (error) {
+    res.status(400).json({ message: error.message }); // 400 Bad Request for validation errors
```

Comment:
Trocar o código de erro para 500, pois ao envolver o código inteiro em um try, o erro pode ocorrer por diversos motivos além de validações. Assim, sempre que o erro for "genérico", adotar o código 500 é uma boa prática para indicar um erro interno no servidor.